src/rewriter/rewrites_template.h

```diff
@@ -30,26 +30,13 @@ class RewriteDb;
  */
 enum class DslProofRule : uint32_t
 {
-  FAIL = 0,
-  REFL,
-  EVAL,
-  // the following rules can be generated temporarily during reconstruction
-  TRANS,
-  CONG,
-  CONG_EVAL,
-  TRUE_ELIM,
-  TRUE_INTRO,
-  ARITH_POLY_NORM,
-  ACI_NORM,
+  NONE = 0,
```

Comment:
I wonder if it's better to replace this tag with `std::optional<DslProofRule>` where it's needed internally. What do you think?

Reply:
I am not in favor of this, since it complicates the logic throughout.  It also would be inconsistent with our other enums e.g. `Kind` which have an error case (`UNDEFINED_KIND`).

Although we should at some point revisit having a consistent style guideline for first and last values of enums.